src/Centreon/Domain/Monitoring/MonitoringService.php

```diff
@@ -359,66 +359,62 @@ public function hidePasswordInCommandLine(Service $monitoringService, string $re
+                true
+            );
+            $onDemandHostMacro = $this->hostConfiguration->findOnDemandHostMacros(
+                $monitoringService->getHost()->getId(),
+                true
+            );
 
-                $configurationToken = explode(' ', $configurationCommand);
-                $monitoringToken = $this->explodeSpacesButKeepValuesByMacro(
-                    $configurationCommand,
-                    $monitoringCommand,
-                    $onDemandServiceMacro,
-                    $onDemandHostMacro
-                );
-                if (count($monitoringToken) === count($configurationToken)) {
-                    /**
-                     * @var array<string, ServiceMacro|HostMacro> $macrosPasswordByName
-                     */
-                    $macrosPasswordByName = [];
-                    foreach ($macrosPassword as $macro) {
-                        $macrosPasswordByName[$macro->getName()] = $macro;
-                    }
-                    $patternMacrosPassword = implode('|', $macroNames);
-                    $patternMacrosPassword = str_replace('$', '\\$', $patternMacrosPassword);
-                    foreach ($configurationToken as $index => $token) {
-                        if (preg_match_all('~' . $patternMacrosPassword . '~', $token, $matches, PREG_SET_ORDER)) {
-                            if (
-                                array_key_exists($matches[0][0], $macrosPasswordByName)
-                                && $macrosPasswordByName[$matches[0][0]]->getValue() !== null
-                            ) {
-                                $monitoringToken[$index] = str_replace(
-                                    $macrosPasswordByName[$matches[0][0]]->getValue(),
-                                    $replacementValue,
-                                    $monitoringToken[$index]
-                                );
-                            }
+            $configurationToken = explode(' ', $configurationCommand);
+            $monitoringToken = $this->explodeSpacesButKeepValuesByMacro(
+                $configurationCommand,
+                $monitoringCommand,
+                $onDemandServiceMacro,
+                $onDemandHostMacro
+            );
+            if (count($monitoringToken) === count($configurationToken)) {
+                /**
+                 * @var array<string, ServiceMacro|HostMacro> $macrosPasswordByName
+                 */
+                $macrosPasswordByName = [];
+                foreach ($macrosPassword as $macro) {
+                    $macrosPasswordByName[$macro->getName()] = $macro;
+                }
+                $patternMacrosPassword = implode('|', $macroNames);
+                $patternMacrosPassword = str_replace('$', '\\$', $patternMacrosPassword);
+                foreach ($configurationToken as $index => $token) {
+                    if (preg_match_all('~' . $patternMacrosPassword . '~', $token, $matches, PREG_SET_ORDER)) {
```

Comment:
Does it fail if password contains `~` ?

Reply:
Yes, I will escape the delimiting character of the regex